extensions/ql-vscode/src/messages.ts

```diff
@@ -487,10 +492,14 @@ export interface UpgradeDescription {
   newSha: string;
 }
 
+
+export type CompiledUpgrades = MultiFileCompiledUpgrades | SingleFileCompiledUpgrade
```

Comment:
And looking at this again, it's not easily clear that these two types differ in only two properties. In this case, I wonder if using proper inheritance would make things clearer. So, something like this:

```js
export interface CompiledUpgrades {
  // add all common properties here
}

interface MultiFileCompiledUpgrades extends CompiledUpgrades {
  newDbscheme: string;
  scripts: CompiledUpgradeScript[];
  compiledUpgradeFile?: never;
}

interface SingleFileCompiledUpgrade extends CompiledUpgrades {
  descriptions: UpgradeDescription[];
  compiledUpgradeFile: string;
}
```

Also. note that I don't think you need to export these types.